pytroll_collectors/tests/test_harvest_schedules.py

```diff
@@ -0,0 +1,257 @@
+
+
+class FakeResponse:
+    def __init__(self, data):
+        self.data = data.split("\n")
+
+    def readlines(self):
+        self.status = 200 if self.data is not None else 404
+        return self.data
+
+    def close(self):
+        pass
+
+
+fake_test_pass_file = """
+EARS-VIIRS Pass Predictions
+2019/12/16 05:08:01
+scheduleBeginEumetsat,scheduleEndEumetsat,Satellite
+2019-12-16 00:24,2019-12-16 00:51,npp
+2019-12-16 01:14,2019-12-16 01:44,noaa20
+2019-12-16 02:05,2019-12-16 02:35,npp
+2019-12-16 02:55,2019-12-16 03:25,noaa20
+2019-12-16 03:46,2019-12-16 04:13,npp
+2019-12-16 04:36,2019-12-16 04:56,noaa20
+2019-12-16 05:26,2019-12-16 05:47,npp
+2019-12-16 06:16,2019-12-16 06:37,noaa20
+2019-12-16 07:05,2019-12-16 07:27,npp
+2019-12-16 07:55,2019-12-16 08:16,noaa20
+2019-12-16 08:46,2019-12-16 09:06,npp
+2019-12-16 09:29,2019-12-16 09:55,noaa20
+2019-12-16 10:17,2019-12-16 10:44,npp
+2019-12-16 11:06,2019-12-16 11:34,noaa20
+2019-12-16 11:57,2019-12-16 12:24,npp
+2019-12-16 12:48,2019-12-16 13:15,noaa20
+2019-12-16 13:37,2019-12-16 14:05,npp
+2019-12-16 14:26,2019-12-16 14:56,noaa20
+2019-12-16 15:18,2019-12-16 15:47,npp
+2019-12-16 16:19,2019-12-16 16:37,noaa20
+2019-12-16 17:10,2019-12-16 17:30,npp
+2019-12-16 18:01,2019-12-16 18:20,noaa20
+2019-12-16 18:55,2019-12-16 19:12,npp
+2019-12-16 19:51,2019-12-16 20:03,noaa20
+2019-12-16 20:42,2019-12-16 20:56,npp
+2019-12-16 21:33,2019-12-16 21:47,noaa20
+2019-12-16 22:24,2019-12-16 22:50,npp
+2019-12-16 23:14,2019-12-16 23:41,noaa20
+"""
+
+
+class TestHarvestSchedules(unittest.TestCase):
+    """Harvest schedule tests."""
+
+    def setUp(self):
+        self.basedir = mkdtemp()
+
+    def tearDown(self):
+        try:
+            shutil.rmtree(self.basedir, ignore_errors=True)
+        except OSError:
+            pass
+
+    def test_pass_list_file_name(self):
+        granule_metadata = {
+            'origin': '157.249.16.188:9063', 'sensor': ['viirs'], 'end_decimal': 3,
+            'stream': 'eumetcast', 'format': 'SDR_compact', 'orig_platform_name': 'npp',
+            'start_time': datetime.datetime(2019, 12, 16, 13, 44, 9), 'variant': 'EARS',
+            'orbit_number': 42154, 'dataset':
+            [{'uri':
+              '/data/pytroll/VIIRS-EARS/SVMC_npp_d20191216_t1344091_e1345333_b42154_c20191216135112000242_eum_ops.h5',
+              'uid': 'SVMC_npp_d20191216_t1344091_e1345333_b42154_c20191216135112000242_eum_ops.h5'}],
+            'start_decimal': 1, 'proctime': '20191216135112000242', 'antenna': 'ears',
+            'data_processing_level': '1B', 'tle_platform_name': 'SUOMI NPP', 'platform_name': 'npp',
+            'end_time': datetime.datetime(2019, 12, 16, 13, 45, 33), 'type': 'HDF5',
+            'collection_area_id': 'eurol'}
+
+        planned_granule_times = set([datetime.datetime(2019, 12, 13, 13, 19),
+                                     datetime.datetime(2019, 12, 13, 13, 38),
+                                     datetime.datetime(2019, 12, 13, 13, 27)])
```

Comment:
This info is used in multiple tests, could they be moved to `setUp` instead?

Reply:
OK